tests/resize_owner.F90

```diff
@@ -0,0 +1,48 @@
+!Copyright 2023 Meteo-France, ECMWF 
+!
+!Licensed under the Apache License, Version 2.0 (the "License");
+!you may not use this file except in compliance with the License.
+!You may obtain a copy of the License at
+!
+!    http://www.apache.org/licenses/LICENSE-2.0
+!
+!    Unless required by applicable law or agreed to in writing, software
+!    distributed under the License is distributed on an "AS IS" BASIS,
+!    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+!    See the License for the specific language governing permissions and
+!    limitations under the License.
```

Comment:
The license header here is missing that one extra sentence we need for ECMWF. @dareg could you please add that in? Same for the other two new files.